### PR TITLE
Add Pull Refresh on Home Screen

### PR DESCRIPTION
**Pull request by wojtus7**
All pull refresh to home screen based on native ScrollView implementation. Pls test it a little because it turned out to be quite tricky to think about all edge cases and handle everything correctly 🫥

iOS and Android:

https://github.com/daimo-eth/daimo/assets/42337257/618caec7-f448-457c-b16d-d79d5508d0e0

https://github.com/daimo-eth/daimo/assets/42337257/1ff53371-425f-4b85-9dec-906bce0dac32

### CODE REVIEW

File: apps/daimo-mobile/src/view/screen/HomeScreen.tsx [hunk 1]
```diff
@@ -63,6 +74,17 @@ function HomeScreenInner({ account }: { account: Account }) {
     keySlot,
     account.homeChainId
   );
+  const [refreshing, setRefreshing] = useState(false);
+
+  const onRefresh = useCallback(async () => {
+    resync("Pull refresh on home screen");
+    setRefreshing(true);
+    await resync("Home screen pull refresh");
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/HomeScreen.tsx:82]: we only have to call resync once, right?

**Review by dcposch — APPROVED** [review]
Video looks fantastic

the diff in OfflineHeader is a big gnarly, but OK to ship.

 left one comment re double `resync()`
 
 afer that, 🚢

File: apps/daimo-mobile/src/view/shared/OfflineHeader.tsx [hunk 2]
```diff
@@ -27,7 +31,7 @@ export function OfflineHeader() {
   return (
     <View style={style}>
       {
-        isAndroid && (
+        isOffline && isAndroid && (
```
[the remark this answers is not in the document]
Comment (wojtus7) [reply apps/daimo-mobile/src/view/shared/OfflineHeader.tsx:34]: I think no because this is only for `isOffline` and the flag is about adding padding when `online` 🤔
